README.md

```diff
@@ -92,6 +93,31 @@ $dns = $factory->createCached('8.8.8.8', $loop, $cache);
 
 See also the wiki for possible [cache implementations](https://github.com/reactphp/react/wiki/Users#cache-implementations).
 
+## Advanced Usage
+
+For more advanced usages one can utilize the `Executor` directly.
```

Comment:
I'm planning to deprecate the `Executor` in future versions, so perhaps we should link to the whole namespace instead (may keep the below example in place though)?